lnwire/dyn_reject.go

```diff
@@ -0,0 +1,73 @@
+package lnwire
+
+import (
+	"bytes"
+	"io"
+)
+
+// DynReject is a message that is sent during a dynamic commitments negotiation
+// process. It is sent by both parties to propose new channel parameters.
+type DynReject struct {
+	// ChanID identifies the channel whose parameters we are trying to
+	// re-negotiate.
+	ChanID ChannelID
+
+	// UpdateRejections is a bit vector that specifies which of the
+	// DynPropose parameters we wish to call out as being unacceptable.
+	UpdateRejections RawFeatureVector
```

Comment:
Do you think that RawFeatureVector should be reimplemented on top of a BitVector? Or should they be separate types altogether. IMO it makes the most sense to refactor RawFeatureVector to use the BitVector structure.

Reply:
In this PR https://github.com/lightningnetwork/lnd/pull/8030, the feature vector is used as a generic bitvector w/o feature bits as well. I think because of the ease of using `RawFeatureVector` as a bitvector, this code doesn't need to change. However, I do think a PR down the line should make `RawFeatureVector` wrap a generic `BitVector` struct we can use.